File: Logitude/Logitude.BL/CommonDataModel/DataContracts/HouseSL.cs
using Logitude.BL.ShipmentsModel.EntityPMs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Logitude.BL.CommonDataModel.DataContracts
{
    public class HouseSL
    {

        public string HouseNumber { get; set; }
        public string ShipmentNumber { get; set; }
        public PartnerSL Shipper  { get; set; }
        public DateTime? HAWBDate { get; set; }
        public string FreightPrepaidCollectId { get; set; }
        public string OtherPrepaidCollectId { get; set; }
        public string MainCarriageCarrierNumber { get; set; }
        public double? ValueOfGoods { get; set; }
        public bool IsDangerous { get; set; }
        
        public string IncotermCode { get; set; }
        public string IncotermName { get; set; }
        public string IncotermId { get; set; }
        public bool IncotermAddedManually { get; set; }


        public string MoveTypeId { get; set; }
        public string MoveTypeName { get; set; }
        public string MoveTypeCode { get; set; }
        public bool MoveTypeAddedManually { get; set; }
        public string MoveTypeTransportModeId { get; set; }


        public string CarrierCode { get; set; }
        public string CarrierName { get; set; }
        public string CarrierId { get; set; }
        public bool CarrierAddedManually { get; set; }


        public string GeneralDescriptionOfGoods { get; set; }


        public string GrossWeightUnitCode { get; set; }
        public string ChargeableWeightUnitCode { get; set; }
        public string VolumeUnitCode { get; set; }
        public string DimensionsUnitCode { get; set; }

        public double? ChargeableWeightInKG { get; set; }
        public bool GrossWeightEdited { get; set; }
        public double? GrossWeightInKG { get; set; }
        public double? GrossWeight { get; set; }
        public double? ChargeableWeight { get; set; }
        public double? TEU { get; set; }
        public int? PackagesQuantity { get; set; }

        public double? Volume { get; set; }
        public double? VolumetricWeight { get; set; }
        public int? NumberOfPackages { get; set; }
        public int? NumberOfContainers { get; set; }

        public double? OrderGrossWeight { get; set; }
        public string ShipmentTypeName { get; set; }
        public string ShipmentTypeId { get; set; }
        public string AgentName { get; set; }

        public string ConsigneeReference1 { get; set; }
        public string ConsigneeReference2 { get; set; }
        public string ShipperReference1 { get; set; }
        public string ShipperReference2 { get; set; }
        public string MainHarmonize { get; set; }



        public string ValueOfGoodsCurrencyId { get; set; }
        public string ValueOfGoodsCurrencyName { get; set; }
        public string ValueOfGoodsCurrencyCode { get; set; }
        public bool ValueOfGoodsCurrencyAddedManually { get; set; }



        public PartnerSL Notify1 { get; set; }
        public PartnerSL Consignee { get; set; }
        private List<ShipmentPackagePM> shipmentPackages;
        public virtual List<ShipmentPackagePM> ShipmentPackages
        {
            get
            {
                if (shipmentPackages == null)
                {
                    shipmentPackages = new List<ShipmentPackagePM>();
                }

                return this.shipmentPackages;
            }

            set
            {
                if (value != null)
                {
                    shipmentPackages = value;
                }
            }
        }

    }
}

File: Logitude/AngularModules/AngularModules/Common/DataContracts/HouseSL.ts
import { PartnerSL } from './PartnerSL';
import {ShipmentPackagePM} from '../../Shipment/EntityPMs/ShipmentPackagePM';
export class HouseSL {
    public HouseNumber: string;
    public ShipmentNumber: string;
    public Shipper: PartnerSL;
    public FreightPrepaidCollectId: string;
    public OtherPrepaidCollectId: string;
    public SharedManifestRef: string;
    public IsDangerous: boolean;

    public IncotermCode: string;
    public IncotermName: string;
    public IncotermId: string;
    public IncotermAddedManually: boolean;
    public ShipperName: string;
    public HAWBDate: Date;
    public CarrierCode: string;
    public CarrierName: string;
    public CarrierId: string;
    public CarrierAddedManually: boolean;
    public MainCarriageCarrierNumber: string;
    public ValueOfGoods: number;
    
    public ConsigneeReference1: string;
    public ConsigneeReference2: string;
    public ShipperReference1: string;
    public ShipperReference2: string;
    public MainHarmonize: string;


    public EntityId: string;
    public Consignee: PartnerSL;
    public Notify1: PartnerSL;
    public Notify2: PartnerSL;

    public GeneralDescriptionOfGoods: string;

    public GrossWeight: number;
    public ChargeableWeight: number;
    public TEU: number;
    public PackagesQuantity: number;

    public Volume: number;
    public VolumetricWeight: number;
    public NumberOfPackages: number;
    public NumberOfContainers: number;
    
    public AgentName: string;

    public  GrossWeightUnitCode : string;
    public  ChargeableWeightUnitCode: string;
    public  VolumeUnitCode : string;
    public  DimensionsUnitCode: string;
    public ChargeableWeightInKG: number;
    public GrossWeightEdited: boolean;
    public  GrossWeightInKG: number;

    public OrderGrossWeight: number;
    public ShipmentTypeId: string;
    public ShipmentTypeName: string;

    public MoveTypeId: string;
    public MoveTypeName: string;
    public  MoveTypeCode: string;
    public MoveTypeAddedManually: boolean;
    public MoveTypeTransportModeId: string;

    public ValueOfGoodsCurrencyId: string;
    public ValueOfGoodsCurrencyName: string;
    public ValueOfGoodsCurrencyCode: string;
    public ValueOfGoodsCurrencyAddedManually: boolean;


    private shipmentPackages: ShipmentPackagePM[];
    get ShipmentPackages() {
        if (this.shipmentPackages == null) {
            this.shipmentPackages = [];
        }

        return this.shipmentPackages;
    }
    set ShipmentPackages(newValue: ShipmentPackagePM[]) {
        if (this.shipmentPackages != newValue) {
            this.shipmentPackages = newValue;
        }
    }
}